gson/src/test/java/com/google/gson/JsonStreamParserTest.java

```diff
@@ -72,57 +72,33 @@ public void testNoSideEffectForHasNext() {
   public void testCallingNextBeyondAvailableInput() {
     JsonElement unused1 = parser.next();
     JsonElement unused2 = parser.next();
-    try {
-      parser.next();
-      fail("Parser should not go beyond available input");
-    } catch (NoSuchElementException expected) {
-    }
+    assertThrows(NoSuchElementException.class, parser::next);
   }
 
   @Test
   public void testEmptyInput() {
     JsonStreamParser parser = new JsonStreamParser("");
-    try {
-      parser.next();
-      fail();
-    } catch (JsonIOException e) {
-      assertThat(e.getCause()).isInstanceOf(EOFException.class);
-    }
+    JsonIOException e = assertThrows(JsonIOException.class, parser::next);
+    assertThat(e.getCause()).isInstanceOf(EOFException.class);
```

Comment:
There is also `assertThat(e).hasCauseThat()`, maybe that would yield better failure messages.
Do you want to use that here and below instead?
```suggestion
    assertThat(e).hasCauseThat().isInstanceOf(EOFException.class);
```

(For the sake of fairness: The old code was also using `assertThat(e.getCause())`, but if you are going to change the code here now anyways, I guess it might be a good time to change this as well.)